mlflow/pytorch/_lightning_autolog.py

```diff
@@ -287,6 +291,136 @@ def on_test_end(self, trainer, pl_module):
         self.metrics_logger.flush()
 
 
+_LATEST_CHECKPOINT_ARTIFACT_TAG_KEY = "_latest_checkpoint_artifact"
+
+
+class __MLflowModelCheckpointCallback(pl.Callback, metaclass=ExceptionSafeAbstractClass):
+
+    def __init__(
+        self,
+        client,
+        run_id,
+        monitor,
+        mode,
+        save_best_only,
+        save_weights_only,
+        save_freq,
+    ):
+        self.client = client
+        self.run_id = run_id
+        self.monitor = monitor
+        self.mode = mode
+        self.save_best_only = save_best_only
+        self.save_weights_only = save_weights_only
+        self.save_freq = save_freq
+        self.latest_checkpoint_timestamp = time.time()
+        self.last_monitor_value = None
+
+    def _is_new_checkpoint_better(self, new_monitor_value):
+        if self.last_monitor_value is None:
+            return True
+
+        if self.mode == "min":
+            return new_monitor_value <= self.last_monitor_value
+
+        if self.mode == "max":
+            return new_monitor_value >= self.last_monitor_value
+
+        assert False, "Illegal __MLflowModelCheckpoint config."
+
+    def _save_checkpoint_rank_zero_only(self, trainer: "pl.Trainer", filepath: str):
+        checkpoint = trainer._checkpoint_connector.dump_checkpoint(self.save_weights_only)
+        trainer.strategy.save_checkpoint(checkpoint, filepath)
+
+    def _check_and_save_checkpoint_if_needed(self, trainer: "pl.Trainer"):
+        current_epoch = trainer.current_epoch
+        metric_dict = {k: float(v) for k, v in trainer.callback_metrics.items()}
+
+        if self.save_best_only:
+            if self.monitor not in metric_dict:
+                # "save-best-only" requires comparing the monitor metric value,
+                # but the provided monitor metric is not available,
+                # skip model checkpoint autologging
+                return
+
+            new_monitor_value = metric_dict[self.monitor]
+            if not self._is_new_checkpoint_better(new_monitor_value):
+                # Current checkpoint is worse than last saved checkpoint,
+                # so skip checkpointing.
+                return
+
+            self.last_monitor_value = new_monitor_value
+
+        if self.save_best_only:
+            if self.save_weights_only:
+                checkpoint_model_filename = "latest_checkpoint.weights.pth"
+            else:
+                checkpoint_model_filename = "latest_checkpoint.pth"
+            checkpoint_metrics_filename = "latest_checkpoint_metrics.json"
+            checkpoint_artifact_dir = "checkpoints"
+        else:
+            if self.save_freq == "epoch":
+                sub_dir_name = f"epoch_{current_epoch}"
+            else:
+                sub_dir_name = f"epoch_{current_epoch}_global_step_{trainer.global_step}"
```

Comment:
we can delete the {current_epoch} part, `trainer.global_step` does not reset after each epoch.